src/RecipeList.tsx

```diff
@@ -0,0 +1,96 @@
+import './RecipeList.css';
+
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+import RecipeCard from './RecipeCard.tsx';
+
+type Recipe = {
+  id: number;
+  name: string;
+  difficulty: string;
+  tags: string[];
+  image: string;
+};
+
+const ITEMS_PER_PAGE = 6;
+
+const RecipeList = () => {
+  const [recipes, setRecipes] = useState<Recipe[]>([]);
+  const [error, setError] = useState('');
+  const [currentPage, setCurrentPage] = useState(1);
+
+  useEffect(() => {
+    type RecipeResponse = {
+      recipes: Recipe[];
+      total: number;
+      skip: number;
+      limit: number;
+    };
+
+    fetch('https://dummyjson.com/recipes')
+      .then((res) => res.json() as Promise<RecipeResponse>)
+      .then((data) => setRecipes(data.recipes))
+      .catch(() => setError('오류'));
+  }, []);
```

Comment:
위 파일이랑 통일성 있게 관리하고 싶으시다면 이렇게 작성하시는 게 더 나을 것 같긴 합니다. 다만 제일 좋은 방법은 api 모듈 자체에서 타입을 관리하는 것이라는 걸 알고 계시면 좋을 거 같아요
```suggestion
    fetch('https://dummyjson.com/recipes')
      .then((res) => res.json())
      .then((data) => setRecipes(data.recipes as RecipeResponse))
      .catch(() => setError('오류'));
  }, []);
```